### PR TITLE
feat: add solution for the 3Sum Closest problem (#16)

### PR DESCRIPTION
**Pull request by mindaugl**
#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by alexpantyukhin — DISMISSED** [review]

**Comment by alexpantyukhin** [comment]
t looks good to me.

File: leetcode/src/16.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+int cmp(const void* a, const void* b) {
+    return *(int*)a - *(int*)b;
+}
+
+int threeSumClosest(int* nums, int nums_size, int target) {        
+    int i, j, k, result, sum3;
+    qsort(nums, nums_size, sizeof(int), cmp);
+    result = nums[0] + nums[1] + nums[2];
+    for (i = 0; i < nums_size-2; i++) {
```
Comment (StepfenShawn) [inline leetcode/src/16.c:11]:
```suggestion
    for (i = 0; i < nums_size - 2; i++) {
```

**Review by StepfenShawn — CHANGES_REQUESTED** [review]

File: leetcode/src/16.c (hunk 1 above)
Comment (mindaugl) [reply leetcode/src/16.c]: Just updated

**Review by StepfenShawn — APPROVED** [review]

File: leetcode/src/16.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#include <stdlib.h> // for qsort()
+
+int cmp(const void* a, const void* b) {
+    return *(int*)a - *(int*)b;
```
Comment (CascadingCascade) [inline leetcode/src/16.c:4]: Actually, comparing two numbers by subtraction is bad practice, the sort will break if the result overflows. [Source](https://stackoverflow.com/a/27284248/918959)
```suggestion
   const int *A = a, *B = b;
   return (*A > *B) - (*A < *B);
```

Comment (mindaugl) [reply leetcode/src/16.c]: Just updated it. For this problem though, overflow could not happen as it is specified that abs(nums[i]) <= 1000.

Comment (CascadingCascade) [reply leetcode/src/16.c]: > Just updated it. For this problem though, overflow could not happen as it is specified that abs(nums[i]) <= 1000.

Yeah, I know that, it's actually one thing I really disliked about those leetcode problems. IRL people input all kinds of stuff into programs, and programmers must have contingency plans for everything or there's going to be a problem. These exercises might be good for familiarizing yourself with the language or various algorithms, but they teach you nothing about best practices that make code more efficient, more maintainable and less bug prone. Competitive programming/Interview questions and actual software engineering are really _very_ different things. By the way, sync your fork and enable actions for it, and here's [some nice reading material](https://sekrit.de/webdocs/c/beginners-guide-away-from-scanf.html)

**Review by StepfenShawn — APPROVED** [review]

**Review by StepfenShawn — DISMISSED** [review]

File: leetcode/src/16.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            } else if (sum3 > target) {
+                k--;
+            } else {
+                return sum3;
+            }
+        }
+    }
+    return result;
+}
+
```
Comment (Panquesito7) [inline leetcode/src/16.c:31]:
```suggestion
```

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by StepfenShawn — APPROVED** [review]
Thanks

**Pull request merged**